analyzers/tests/SonarAnalyzer.UnitTest/Extensions/BlockSyntaxExtensionsTests.cs

```diff
@@ -0,0 +1,316 @@
+            if (isEmpty)
+            {
+                declaration.Body.Should().BeEmpty(treatCommentsAsContent, treatConditionalCompilationAsContent);
+            }
+            else
+            {
+                declaration.Body.Should().BeNotEmpty(treatCommentsAsContent, treatConditionalCompilationAsContent);
+            }
+        }
+
+        [TestMethod]
+        public void IsEmpty_TrailingTriviaOnOpenBrace_Comment()
+        {
+            var block = BlockWithoutTrivia();
+            block = block.WithOpenBraceToken(Token(SyntaxKind.OpenBraceToken).WithTrailingTrivia(ParseTrailingTrivia("// Comment")));
+            block.Should().BeNotEmpty(treatCommentsAsContent: true);
+        }
+
+        [TestMethod]
+        public void IsEmpty_LeadingTriviaOnCloseBrace_Comment()
+        {
+            var block = BlockWithoutTrivia();
+            block = block.WithCloseBraceToken(Token(SyntaxKind.CloseBraceToken).WithLeadingTrivia(ParseLeadingTrivia("// Comment")));
+            block.Should().BeNotEmpty(treatCommentsAsContent: true);
+        }
+
+        [TestMethod]
+        public void IsEmpty_TrailingTriviaOnOpenBrace_ConditionalCompilation()
+        {
+            var block = BlockWithoutTrivia();
+            block = block.WithOpenBraceToken(Token(SyntaxKind.OpenBraceToken).WithTrailingTrivia(ParseTrailingTrivia("#if Something\r\n var i = 1; \r\n #endif")));
+            block.Should().BeNotEmpty(treatConditionalCompilationAsContent: true);
+        }
+
+        [TestMethod]
+        public void IsEmpty_LeadingTriviaOnCloseBrace_ConditionalCompilation()
+        {
+            var block = BlockWithoutTrivia();
+            block = block.WithCloseBraceToken(Token(SyntaxKind.CloseBraceToken).WithLeadingTrivia(ParseLeadingTrivia("#if Something\r\n var i = 1; \r\n #endif")));
+            block.Should().BeNotEmpty(treatConditionalCompilationAsContent: true);
+        }
+
+        private static BlockSyntax BlockWithoutTrivia()
+        {
+            var block = MethodDeclarationForBlock("{}").Body;
+            block = block
+                .WithOpenBraceToken(block.OpenBraceToken.WithoutTrivia())
+                .WithCloseBraceToken(block.CloseBraceToken.WithoutTrivia());
+            return block;
+        }
+
+        private static MethodDeclarationSyntax MethodDeclarationForBlock(string methodBlock)
+            => MethodDeclaration(WrapInClass(methodBlock));
+
+        private static string WrapInClass(string methodBlockOrArrow) => $@"
+public class C
+{{
+    public void M()
+    {methodBlockOrArrow}
+}}";
+
+        private static MethodDeclarationSyntax MethodDeclaration(string source)
+        {
+            var root = CSharpSyntaxTree.ParseText(source).GetRoot();
+            root.ContainsDiagnostics.Should().BeFalse();
+            return root.DescendantNodes().OfType<MethodDeclarationSyntax>().Single();
+        }
+    }
+
+    internal static class BlockSyntaxExtensions
```

Comment:
I think we can move the assertions outside of this test file. They can be used in other tests as well.

Reply:
I was thinking about removing the BlockSyntaxAssertions class. The reasons are:
* There is no need to have "IsEmpty" and "NotIsEmpty" as public extension methods. "IsEmpty" is enough.
* If there is only one extension method, we don't need to assert "IsEmpty" and "IsNotEmpty" in one go (one of the main benefits of the assertion class here).
* It is odd to have the actual SUT in the assertion class. This is not what these custom assertions are meant for.
* The assertion class is not reusable. First of all, it calls "IsEmpty" and "IsNotEmpty" which is not needed outside this test file. It also doesn't bring much value: `block.IsEmpty().Should().BeTrue()` vs `block.Should().BeEmpty()` is not much of a difference.

What do you think?

Reply:
Sounds good to me.